packages/embeds/embed-react/src/index.ts

```diff
@@ -1,14 +1,20 @@
 "use client";
 
-import type { GlobalCal } from "@calcom/embed-core";
+import type { GlobalCal, GlobalCalWithoutNs } from "@calcom/embed-core";
 import EmbedSnippet from "@calcom/embed-snippet";
 
 import Cal from "./Cal";
 
-export const getCalApi = (embedJsUrl?: string): Promise<GlobalCal> =>
-  new Promise(function tryReadingFromWindow(resolve) {
-    EmbedSnippet(embedJsUrl);
-    const api = window.Cal;
+export const getCalApi = (options?: {
```

Comment:
We are changing the API here in a breaking fashion. 

Users who are using older version of embed-react, would use the old API(but that won't support namespacing) and those moving to new versions would have to use the new API which supports namespacing.